src/luxonis_ml/embeddings/emb_mistakes.py

```diff
@@ -0,0 +1,140 @@
+"""
+Mismatch Detection in Labelled Data
+
+This module provides functionalities to detect mismatches or potential mislabelling 
+in a dataset based on various strategies. This is crucial in supervised machine 
+learning tasks where the quality of labels significantly affects model performance.
+
+Methods implemented:
+    - Centroids: This method identifies mismatches by comparing the distance of data points 
+                 to the centroid of their own class against the distances to centroids of 
+                 other classes.
+    - KNN (k-Nearest Neighbors): This approach leverages the idea that if the majority of data 
+                                 is correctly labelled, then mislabelled data will be corrected 
+                                 by its nearest neighbors.
+    - [Note: DBSCAN was considered but not implemented due to underperformance.]
+
+Usage:
+    To use this module, import the desired methods and provide the embeddings and labels:
+    ```python
+    from mismatch_detection import find_mismatches_centroids, find_mismatches_knn
+
+    # Detect mismatches using centroids
+    mismatches, new_labels = find_mismatches_centroids(X_train, y_train)
+    
+    # Detect mismatches using KNN
+    mismatches, new_labels = find_mismatches_knn(X_train, y_train)
+    ```
+"""
+
+import numpy as np
+
+import scipy.spatial.distance as distance
+from sklearn.neighbors import KNeighborsClassifier
+
+from luxonis_ml.embeddings.qdrant_utils import *
+
+
+def find_mismatches_centroids(X, y):
+    """
+    Find mismatches in the dataset. 
+    A mismatch is defined as a sample that is closer to another centroid than to its own centroid.
+
+    Parameters
+    ----------
+    X : np.array
+        The embeddings to use.
+    y : np.array
+        The targets to use.
```

Comment:
More of a question, but is `y` assumed to always be a single label per image? Does this method only apply to labeled data for image classification?

Reply:
Yes, currently its implemented like that. Should it support any other cv tasks?

Reply:
Maybe one day but I don't think so right now - I was just curious